src/compute-client/src/plan/mod.rs

```diff
@@ -1093,11 +1115,29 @@ impl<T: timely::progress::Timestamp> Plan<T> {
 
                         let forms = AvailableCollections::new_raw();
 
-                        lir_value = Plan::ArrangeBy {
-                            input: Box::new(lir_value),
-                            forms,
-                            input_key,
-                            input_mfp,
+                        // We just want to insert an `ArrangeBy` to form an unarranged collection,
+                        // but there is a complication: We shouldn't break the invariant (created by
+                        // `NormalizeLets`, and relied upon by the rendering) that there isn't
+                        // anything between two `LetRec`s. So if `lir_value` is itself a `LetRec`,
+                        // then we insert the `ArrangeBy` on the `body` of the inner `LetRec`,
+                        // instead of on top of the inner `LetRec`.
+                        lir_value = match &lir_value {
```

Comment:
I think you can `match lir_value` instead of `match &lir_value` and avoid the `{ids, values, body}.clone()` calls (you'll need to rename `_non_let_rec` to `lir_value`).

Reply:
Hmm, yes, thanks! I've changed it!